db/set.go

```diff
@@ -116,3 +220,158 @@ func (set *Set) SMembers() ([][]byte, error) {
 	}
 	return members, nil
 }
+
+// SCard returns the set cardinality (number of elements) of the set stored at key
+func (set *Set) SCard() (int64, error) {
+	if !set.Exists() {
+		return 0, nil
+	}
+	return set.meta.Len, nil
+}
+
+// SIsmember returns if member is a member of the set stored at key
+func (set *Set) SIsmember(member []byte) (int64, error) {
+	if !set.Exists() {
+		return 0, nil
+	}
+	dkey := DataKey(set.txn.db, set.meta.ID)
+	ikey := setItemKey(dkey, member)
+
+	value, err := set.txn.t.Get(ikey)
+	if err != nil {
+		if IsErrNotFound(err) {
+			return 0, nil
+		}
+		return 0, err
+	}
+	if !bytes.Equal(value, SetNilValue) {
+		return 0, nil
+	}
+	return 1, nil
+}
+
+// SPop removes and returns one or more random elements from the set value store at key.
+func (set *Set) SPop(count int64) (members [][]byte, err error) {
+	if !set.Exists() || set.meta.Len == 0 {
+		return nil, nil
+	}
+	dkey := DataKey(set.txn.db, set.meta.ID)
+	prefix := append(dkey, ':')
+	iter, err := set.txn.t.Iter(prefix, nil)
+	if err != nil {
+		return nil, err
+	}
+	defer iter.Close()
+	var del int64
+	var ms [][]byte
+	for iter.Valid() && iter.Key().HasPrefix(prefix) {
+		if count == 0 {
+			ms = append(ms, iter.Key()[len(prefix):])
+			if err := set.txn.t.Delete([]byte(iter.Key())); err != nil {
+				return nil, err
+			}
+			set.meta.Len--
+			break
+		} else {
```

Comment:
The loop will break when `count == 0`, so the `else` statement is not necessary. Remove it then the code could be more cleaner.

Reply:
yes,you are right